### DISCUSSION TITLE
Double Splash Screen on IOS

### DISCUSSION THREAD

**Discussion by warisanwar** [timbru31/cordova-plugin-lottie-splashscreen#465] [General]
Hello,
I'm trying to use the library on my ionic capacitor project, I'm able to display the splash screen but the issue is that black screen with a white container is displaying first followed by the splash screen.

![image](https://user-images.githubusercontent.com/28192108/181450649-08ca6d86-8784-4aac-b9a2-8307dad9c28e.png)

I tried debugging the issue on xcode, and saw that LottieSplashScreen is being invoked twice.

<img width="656" alt="Screenshot 2022-07-28 at 10 36 26 AM" src="https://user-images.githubusercontent.com/28192108/181448421-0b0e4d0c-fb7d-48ef-a062-6a77b47b197b.png">

This is how I've configured my `capacitor.config.ts` file:

`import { CapacitorConfig } from '@capacitor/cli';

const config: CapacitorConfig = {
  appId: 'com.abcd.ios',
  appName: 'abcd',
  webDir: 'dist/abcd-app',
  bundledWebRuntime: false,
  // @ts-ignore
  SplashScreen: {
    launchAutoHide: true,
    launchShowDuration: 0,
  },
  cordova: {
    preferences: {
      "LottieHideAfterAnimationEnd": "true",
      "LottieFullScreen": "true",
    }
  }
};

export default config;`

And this is my `app.component.ts` file:

<img width="846" alt="Screenshot 2022-07-28 at 10 45 18 AM" src="https://user-images.githubusercontent.com/28192108/181450304-397a0ce7-9b03-4e64-954c-529239a22319.png">

**Comment by warisanwar** [comment]
To fix this issue, we removed the line `<param name="onload" value="true"/>`

```
<feature name="LottieSplashScreen">
  <param name="ios-package" value="LottieSplashScreen"/>
  <param name="onload" value="true"/>      <----------------Remove this line
</feature>
```

**in ios/App/App/config.xml**

**Comment by timbru31** [comment]
The whole idea of the lottie splashscreen is that it's loaded before your JS code is run. Normally I'd recommend to not invoke the `.show()` method as this is done automatically and just use `.hide()` once you've deemed your app is fully loaded.

There might be use cases (such as an auto update) where you'd need to show a splash again to avoid a flickering/white screen on reload of the webview.

Of course your workaround also works by disabling the initial onload call.
